### PR TITLE
bpo-32718: Make Activate.ps1 for venv cross-platform and available on all platforms

### PR DESCRIPTION
**Pull request by brettcannon**
PowerShell Core 6.1 is the cross-platform port of Windows PowerShell. This change updates Activate.ps1 to not make Windows assumptions as well as installing it into the bin/Scripts directory on all operating systems.

Requires PowerShell Core 6.1 for proper readline support once the shell has been activated for the virtual environment.

### CODE REVIEW

File: Lib/venv/scripts/common/Activate.ps1 [hunk 1]
```diff
@@ -48,4 +53,4 @@ if (Test-Path env:PYTHONHOME) {
 
 # Add the venv to the PATH
 copy-item env:PATH env:_OLD_VIRTUAL_PATH
-$env:PATH = "$env:VIRTUAL_ENV\__VENV_BIN_NAME__;$env:PATH"
+$env:PATH = add-bin $env:PATH
```
Comment (d3r3kk) [inline Lib/venv/scripts/common/Activate.ps1:56]: Another way you can do this, without having to create a function, could be like so:

```powershell
$env:PATH = "$(Join-Path -Path $env:VIRTUAL_ENV -ChildPath '__VENV_BIN_NAME__');$env:PATH"
```

...but having said that, I prefer the way you've done it here.

File: Lib/venv/scripts/common/Activate.ps1 [hunk 2]
```diff
@@ -1,3 +1,8 @@
+function Script:add-bin($envPath) {
```
Comment (d3r3kk) [inline Lib/venv/scripts/common/Activate.ps1]: Feel free to stick type declarations in PowerShell.

```powershell
function Script:add-bin([string]$envPath) { ...
```

**Review by d3r3kk — APPROVED** [review]
Nice!

**Comment by d3r3kk** [comment]
BTW: I tested the script out on my Ubuntu 18.04 machine and it works like a charm.

**Review by zooba — APPROVED** [review]

**Pull request merged**